doc/style_guide.md

```diff
@@ -116,6 +116,12 @@ Code style rules
   runnable scripts they need to begin with a shebang and be marked
   executable.
 
+- Defensive programming is recommended. The idea is to write code that
+  expects other parts of the codebase to mess up somewhere. For example,
+  when checking if a species population is extinct, instead of checking
```

Comment:
`species population is extinct` sounds a bit weird to me. I'd just drop the population word from there `species is extinct`